homeassistant/components/synology_dsm/__init__.py

```diff
@@ -144,30 +247,230 @@ async def async_setup(self):
             device_token=self._entry.data.get("device_token"),
         )
 
+        await self._async_should_fetch_api()
+
         await self._hass.async_add_executor_job(self._fetch_device_configuration)
-        await self.update()
+        await self.async_update()
 
         self._unsub_dispatcher = async_track_time_interval(
             self._hass,
-            self.update,
+            self.async_update,
             timedelta(
                 minutes=self._entry.options.get(
                     CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
                 )
             ),
         )
 
+    @callback
+    def subscribe(self, api_key, unique_id):
+        """Subscribe an entity from API fetches."""
+        if api_key not in self._fetching_entities:
+            self._fetching_entities[api_key] = set()
+        self._fetching_entities[api_key].add(unique_id)
+
+        @callback
+        def unsubscribe() -> None:
+            """Unsubscribe an entity from API fetches (when disable)."""
+            self._fetching_entities[api_key].remove(unique_id)
+
+        return unsubscribe
+
+    async def _async_should_fetch_api(self):
```

Comment:
```suggestion
   @callback
   def _async_should_fetch_api(self):
```

Nothing is being awaited here so this can be a callback

Reply:
Maybe rename to `_async_setup_api_requests` since nothing is being returned here